### DISCUSSION TITLE
difference between component_{id,name}

### DISCUSSION THREAD

**Discussion by tshepang** [vectordotdev/vector#14079] [Q&A] [answered]
I was thinking these are interchangeable names, but I see this line on log:
> ERROR transform{component_kind="transform" component_id=node-activity-to-metric component_type=log_to_metric component_name=node-activity-to-metric}...

**Comment by jszwedko** [comment, marked as answer]
Hi @tshepang! `component_name` is the old field name and has been deprecated for `component_id`. `component_name` will be dropped at some point in the future.
